.github/workflows/run-test-n-times.yml

```diff
@@ -0,0 +1,141 @@
+name: Run Test N-times
+
+on:
+  workflow_dispatch:
+    inputs:
+      commit:
+        description: "Commit SHA"
+        required: true
+      n_runs:
+        description: "Number of times to repeat the test"
+        required: true
+      test_name:
+        description: "Name of the test suite to run, include -testify.m flag here if desired (i.e. 'TestAcquireShard_DeadlineExceededErrorSuite' or 'TestFunctionalSuite -testify.m=TestUpdateWorkflow')"
+        required: true
+
+env:
+  COMMIT: ${{ github.event.inputs.commit || github.sha }}
+  DOCKER_COMPOSE_FILE: ./develop/github/docker-compose.yml
+  TEMPORAL_VERSION_CHECK_DISABLED: 1
+  BUILDKITE_ANALYTICS_TOKEN: ${{ secrets.BUILDKITE_ANALYTICS_TOKEN }}
+
+jobs:
+  misc-checks:
+    name: Misc checks
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-20.04]
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ env.COMMIT }}
+          submodules: true
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          check-latest: true
+
+      - uses: arduino/setup-protoc@v3
+
+      - run: make ci-build-misc
+
+      - run: make build-tests
```

Comment:
we may only need the `make build-tests` part of this, I just wanted to leave this whole Misc checks step as is since I'm not sure what side-effects are needed for the next steps